### ISSUE TITLE
Brazilian Portuguese: UI remains in English

### ISSUE THREAD

**Issue by mlep** [JabRef/jabref#1182]
JabRef version JabRef-3.3dev--snapshot--2016-04-12--master--f8bce59.jar on Linux Mint LMDE

Steps to reproduce:
1. Options -> Preferences
2. Select "Brazilian Portuguese"
3. Restart JabRef
   ==> English interface...
   (all other languages work as expected)

Log:

22:03:37.907 [AWT-EventQueue-0] WARN  net.sf.jabref.logic.l10n.Localization - Could not find bundles for language pt_br, switching to full english language
java.util.MissingResourceException: Can't find bundle for base name l10n/JabRef, locale pt_br
    at java.util.ResourceBundle.throwMissingResourceException(ResourceBundle.java:1564) ~[?:1.8.0_66]
    at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1387) ~[?:1.8.0_66]
    at java.util.ResourceBundle.getBundle(ResourceBundle.java:890) ~[?:1.8.0_66]
    at net.sf.jabref.logic.l10n.Localization.createResourceBundles(Localization.java:41) ~[JabRef-3.3dev--snapshot--2016-04-12--master--f8bce59.jar:?]
    at net.sf.jabref.logic.l10n.Localization.setLanguage(Localization.java:32) ~[JabRef-3.3dev--snapshot--2016-04-12--master--f8bce59.jar:?]
    at net.sf.jabref.JabRef.start(JabRef.java:84) ~[JabRef-3.3dev--snapshot--2016-04-12--master--f8bce59.jar:?]
    at net.sf.jabref.JabRefMain.lambda$main$0(JabRefMain.java:10) ~[JabRef-3.3dev--snapshot--2016-04-12--master--f8bce59.jar:?]
    at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311) [?:1.8.0_66]
    at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:756) [?:1.8.0_66]
    at java.awt.EventQueue.access$500(EventQueue.java:97) [?:1.8.0_66]
    at java.awt.EventQueue$3.run(EventQueue.java:709) [?:1.8.0_66]
    at java.awt.EventQueue$3.run(EventQueue.java:703) [?:1.8.0_66]
    at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_66]
    at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:76) [?:1.8.0_66]
    at java.awt.EventQueue.dispatchEvent(EventQueue.java:726) [?:1.8.0_66]
    at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201) [?:1.8.0_66]
    at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116) [?:1.8.0_66]
    at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105) [?:1.8.0_66]
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101) [?:1.8.0_66]
    at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93) [?:1.8.0_66]
    at java.awt.EventDispatchThread.run(EventDispatchThread.java:82) [?:1.8.0_66]
Caused by: java.lang.NullPointerException
    at java.io.Reader.<init>(Reader.java:78) ~[?:1.8.0_66]
    at java.io.InputStreamReader.<init>(InputStreamReader.java:113) ~[?:1.8.0_66]
    at net.sf.jabref.logic.l10n.EncodingControl.newBundle(EncodingControl.java:51) ~[JabRef-3.3dev--snapshot--2016-04-12--master--f8bce59.jar:?]
    at java.util.ResourceBundle.loadBundle(ResourceBundle.java:1501) ~[?:1.8.0_66]
    at java.util.ResourceBundle.findBundle(ResourceBundle.java:1465) ~[?:1.8.0_66]
    at java.util.ResourceBundle.getBundleImpl(ResourceBundle.java:1361) ~[?:1.8.0_66]
    ... 19 more

**Comment by Siedlerchr** [comment]
I think it is just a Case-Sensitive filename problem. I will look at that

**Comment by gutierrezps** [comment]
I'm using 3.8.2, and I'm facing the same problem. I think it's the same case issue.

Log:

```
12:45:11.153 [AWT-EventQueue-0] WARN  net.sf.jabref.logic.l10n.Localization - Could not find bundles for language pt_br, switching to full english language
java.util.MissingResourceException: Can't find bundle for base name l10n/JabRef, locale pt_br
	at java.util.ResourceBundle.throwMissingResourceException(Unknown Source) ~[?:1.8.0_131]
	at java.util.ResourceBundle.getBundleImpl(Unknown Source) ~[?:1.8.0_131]
	at java.util.ResourceBundle.getBundle(Unknown Source) ~[?:1.8.0_131]
	at net.sf.jabref.logic.l10n.Localization.createResourceBundles(Localization.java:44) ~[JabRef-3.8.2.jar:?]
	at net.sf.jabref.logic.l10n.Localization.setLanguage(Localization.java:35) ~[JabRef-3.8.2.jar:?]
	at net.sf.jabref.JabRefMain.start(JabRefMain.java:56) ~[JabRef-3.8.2.jar:?]
	at net.sf.jabref.JabRefMain.lambda$main$0(JabRefMain.java:40) ~[JabRef-3.8.2.jar:?]
	at java.awt.event.InvocationEvent.dispatch(Unknown Source) [?:1.8.0_131]
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source) [?:1.8.0_131]
	at java.awt.EventQueue.access$500(Unknown Source) [?:1.8.0_131]
	at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_131]
	at java.awt.EventQueue$3.run(Unknown Source) [?:1.8.0_131]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:1.8.0_131]
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source) [?:1.8.0_131]
	at java.awt.EventQueue.dispatchEvent(Unknown Source) [?:1.8.0_131]
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source) [?:1.8.0_131]
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source) [?:1.8.0_131]
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source) [?:1.8.0_131]
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_131]
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source) [?:1.8.0_131]
	at java.awt.EventDispatchThread.run(Unknown Source) [?:1.8.0_131]
Caused by: java.lang.NullPointerException
	at java.io.Reader.<init>(Unknown Source) ~[?:1.8.0_131]
	at java.io.InputStreamReader.<init>(Unknown Source) ~[?:1.8.0_131]
	at net.sf.jabref.logic.l10n.EncodingControl.newBundle(EncodingControl.java:51) ~[JabRef-3.8.2.jar:?]
	at java.util.ResourceBundle.loadBundle(Unknown Source) ~[?:1.8.0_131]
	at java.util.ResourceBundle.findBundle(Unknown Source) ~[?:1.8.0_131]
	... 20 more
```

**Comment by koppor** [comment]
Operating system? How do you start JabRef?

**Comment by gutierrezps** [comment]
I'm running the standalone jar by double clicking on it. Windows 10 x64, Java 8 update 131.

**Comment by rscaputo** [comment]
Same problem here with Jabref 3.8.2 Win10x64, Languge UI doesn't change after language is setup to Brazilian Portuguese in preferences.

**Comment by Siedlerchr** [comment]
In the lastest snapshot of JabRef 4.x I can't reproduce this. Due to some missing translations, there are still menu items in English, but the rest looks translated to me:
![grafik](https://user-images.githubusercontent.com/320228/27970982-4f00c822-6351-11e7-9282-f887fcc44f11.png)

**Comment by rscaputo** [comment]
@Siedlerchr updated to jabref 4 beta and the problem persisted.

JabRef 4.0.0-beta
Windows 10 10.0 amd64 
Java 1.8.0_131

Steps to reproduce:

1. Installed JabRef 4.0.0-beta
2. Changed Language to Brazilian Portuguese in Preferences
3. Restarted Jabref
4. Language Didn't change
5. Other languages worked fine.

<details>
<summary>Detail information:</summary>

```
Could not find bundles for language pt_br, switching to full english language
java.util.MissingResourceException: Can't find bundle for base name l10n/JabRef, locale pt_br
	at java.util.ResourceBundle.throwMissingResourceException(Unknown Source)
	at java.util.ResourceBundle.getBundleImpl(Unknown Source)
	at java.util.ResourceBundle.getBundle(Unknown Source)
	at org.jabref.logic.l10n.Localization.createResourceBundles(Localization.java:52)
	at org.jabref.logic.l10n.Localization.setLanguage(Localization.java:43)
	at org.jabref.JabRefMain.start(JabRefMain.java:69)
	at org.jabref.JabRefMain.lambda$start$0(JabRefMain.java:53)
	at java.awt.event.InvocationEvent.dispatch(Unknown Source)
	at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
	at java.awt.EventQueue.access$500(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.awt.EventQueue$3.run(Unknown Source)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(Unknown Source)
	at java.awt.EventQueue.dispatchEvent(Unknown Source)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
	at java.awt.EventDispatchThread.run(Unknown Source)
Caused by: java.lang.NullPointerException
	at java.io.Reader.<init>(Unknown Source)
	at java.io.InputStreamReader.<init>(Unknown Source)
	at org.jabref.logic.l10n.EncodingControl.newBundle(EncodingControl.java:51)
	at java.util.ResourceBundle.loadBundle(Unknown Source)
	at java.util.ResourceBundle.findBundle(Unknown Source)
	... 20 more

```

</details>

**Comment by Siedlerchr** [comment]
Okay, thanks for your test. I could reproduce this now when starting the jar directly or by using java -jar JabRef...jar 
Because running from gradle/Eclipse was okay:
Theoretically this should have been fixed in 
https://github.com/JabRef/jabref/commit/85e71eacbfb9ba0866f920a786c1813564f26845

**Comment by Siedlerchr** [comment]
Could you please try the latest snapshot?
http://builds.jabref.org/master/

**Comment by rscaputo** [comment]
Unlike the other versions I installed before, this snapshot is not being allowed to install by my antivirus...

Antivirus: Panda Protection V. 18.01.00
Detected virus W32/Exploit.gen

**Comment by Siedlerchr** [comment]
Thanks for your feedback. That's a false positive, I did a scan with Virustotal both 64 bit and 432bit.
https://www.virustotal.com/de/file/98151afd7b241e7594ada8bb467ac4d388a1a101ca33651dcd315be175cae8af/analysis/1499623453/

https://www.virustotal.com/de/file/408f089a02058ea8f8b279186e4f5a93d34e9440320af515ced90ea1000e2322/analysis/1499623163/

I don't know if you have an option to white list the application. 
@koppor Maybe we need to concact the people at Panda to whitellist JabRef? 

Anyway I close this bug as I tested it with the current snapshot and it worked fine.

As wirtten earlier, we would be happy if you (or someone else from the Brazilian Portugese speaking users) could [contribute to the translation](https://help.jabref.org/en/TranslatingGUI) of the user Interface.
